plugins/main/public/components/overview/office/dashboard/dashboard.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  AlertsDataSourceRepository,
+  PatternDataSource,
+  tParsedIndexPattern,
+  useDataSource,
+} from '../../../common/data-source';
+import { Office365DataSource } from '../../../common/data-source/pattern/alerts/office-365/office-365-data-source';
+import { DiscoverNoResults } from '../../../common/no-results/no-results';
+import { LoadingSpinner } from '../../../common/loading-spinner/loading-spinner';
+
+const plugins = getPlugins();
+
+const SearchBar = getPlugins().data.ui.SearchBar;
+
+const DashboardByRenderer = plugins.dashboard.DashboardContainerByValueRenderer;
+
+const DashboardOffice365Component: React.FC = () => {
+  const {
+    filters,
+    dataSource,
+    fetchFilters,
+    isLoading: isDataSourceLoading,
+    fetchData,
+    setFilters,
+  } = useDataSource<tParsedIndexPattern, PatternDataSource>({
+    DataSource: Office365DataSource,
+    repository: new AlertsDataSourceRepository(),
+  });
+
+  const [results, setResults] = useState<SearchResponse>({} as SearchResponse);
+  const { searchBarProps } = useSearchBar({
+    indexPattern: dataSource?.indexPattern as IndexPattern,
+    filters,
+    setFilters,
+  });
+  const { query } = searchBarProps;
+
+  useEffect(() => {
+    if (isDataSourceLoading) {
+      return;
+    }
+    fetchData({
+      query,
+      dateRange: {
+        from: searchBarProps.dateRangeFrom || '',
+        to: searchBarProps.dateRangeTo || '',
+      },
+    })
+      .then(results => {
+        setResults(results);
+      })
+      .catch(error => {
+        const searchError = ErrorFactory.create(HttpError, {
+          error,
+          message: 'Error fetching vulnerabilities',
+        });
+        ErrorHandler.handleError(searchError);
+      });
+  }, [
+    JSON.stringify(fetchFilters),
+    JSON.stringify(query),
+    searchBarProps.dateRangeFrom,
+    searchBarProps.dateRangeTo,
+  ]);
+
+  return (
+    <>
+      <I18nProvider>
+        <>
+          {isDataSourceLoading && !dataSource ? (
+            <LoadingSpinner />
+          ) : (
+              <div className='wz-search-bar'>
+                <SearchBar
+                  appName='google-cloud-searchbar'
+                  {...searchBarProps}
+                  showDatePicker={true}
+                  showQueryInput={true}
+                  showQueryBar={true}
+                  showSaveQuery={true}
+                />
+              </div>
+            )}
+          {dataSource && results?.hits?.total === 0 ? (
+            <DiscoverNoResults />
+          ) : null}
+          {dataSource && results?.hits?.total > 0 ? (
+            <div className='office-365-dashboard-responsive'>
+              <SampleDataWarning />
+              <DashboardByRenderer
+                input={{
+                  viewMode: ViewMode.VIEW,
+                  panels: getKPIsPanel(dataSource.id),
+                  isFullScreenMode: false,
+                  filters: searchBarProps.filters ?? [],
```

Comment:
Shouldn't the filters attribute receive the `fetchFilters`?

Reply:
Solved in https://github.com/wazuh/wazuh-dashboard-plugins/pull/6528/commits/5f60571ef95a2fb4f255c1301d36c59cfb270df7